script.js

```diff
@@ -1,12 +1,442 @@
+        );
+      }
+    })
+    .then((data) => {
+      allEpisodes = data;
+      showSearchInputHtml.value = "";
+      makePageForEpisodes(allEpisodes);
+      buildEpisodeDropdownList(allEpisodes);
+    })
+    .catch((error) => {
+      console.log(error);
+    });
+}
+
+// level 100
+
+/*
+  If the user clicks show select then the show select will disappear
+  If the user clicks the see show listing button then the episodes will disappear and the shows will appear also the episode list should not be shown initially
+*/
+
+function toggleVisibility(showElement, hideElement) {
+  showElement.style.display = "inline"; // makes the element visible
+  hideElement.style.display = "none"; // hides the element
+}
+
+function getFormattedEpisodeNameInDropdown(episode) {
+  const episodeTitle = episode.name;
+  const paddedSeasonNum = ("0" + episode.season).slice(-2);
+  const paddedEpisodeNum = ("0" + episode.number).slice(-2);
+  return `S${paddedSeasonNum}E${paddedEpisodeNum} - ${episodeTitle}`;
+}
+
+function getFormattedEpisodeNameInEpisodeCards(episode) {
+  const paddedSeasonNum = ("0" + episode.season).slice(-2);
+  const paddedEpisodeNum = ("0" + episode.number).slice(-2);
+  return `${episode.name} - S${paddedSeasonNum}E${paddedEpisodeNum}`;
 }
 
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  // make the episode dropdown list visible and hide the show dropdown list
+  toggleVisibility(episodeSelectHtml, showSelectHtml);
+  // make the episode searchbar visible and hide the show searchbar from show page
+  toggleVisibility(episodeSearchInputHtml, showSearchInputHtml);
+  // make the episode search result visible and hide the show search result
+  toggleVisibility(episodeSearchResultDisplayHtml, showSearchResultDisplayHtml);
+
+  showSearchInputHtml.removeEventListener("input", searchForShow);
+  episodeSearchInputHtml.addEventListener("input", searchForEpisode);
+
+  rootHtml.innerHTML = "";
+  const cardsContainerHtml = document.createElement("div"); // Creating the cards container element inside the rootEl
+
+  cardsContainerHtml.classList.add("episode-cards-container");
+
+  // Loop through each episode and create a card for it
+
+  episodeList.forEach((episode, index) => {
+    // Get the episode name, season number, episode number, and combine them using the right function
+    const episodeNameNumSeasonCombo = getFormattedEpisodeNameInEpisodeCards(episode);
+
+    // Create the episode card elements
+
+    const episodeCardHtml = document.createElement("div");
+    episodeCardHtml.classList.add("episode-card");
+    episodeCardHtml.id = "episode-card" + index;
+
+    // Get the episode title
+
+    const episodeNameNumSeasonComboHolderHtml = document.createElement("div");
+    const episodeNameNumSeasonComboTextHtml = document.createElement("span");
+    episodeNameNumSeasonComboHolderHtml.classList.add(
```

Comment:
It looks like every element you create, you add a class-name for - can you maybe extract a function which would mean you could write something like:
```js
const episodeNameNumSeasonComboHolderHtml = createElement("div", "episode-name-num-holder");
```

instead of keep having to the same things over and over?

Are there more things you're repeatedly doing you could _also_ extract into this function, as well?